src/main/java/com/codefellowship/codefellowship/controllers/AppUserController.java

```diff
@@ -73,5 +74,52 @@ public void authWithHttpServletRequest(String username, String password) {
         }
     }
 
+    @GetMapping("/test")
+    public String getTestPage(Model m, Principal p) {
+        if(p != null) {
+            String username = p.getName();
+            AppUser user = appUserRepo.findByUsername(username);
+            if(user != null) {
+                m.addAttribute("username", user.getUsername());
+            }
+        }
+        return "test.html"
+    }
+
+    @GetMapping("/user/{id}")
+    public String getUserInfoPage(Model m, Principal p, @PathVariable long id) {
+        if (p != null) {
+            String username = p.getName();
+            AppUser browsing = appUserRepo.findByUsername(username);
+            m.addAttribute("username", browsingUser.getUsername());
+        }
+        AppUser profileUser = appUserRepo.findById(id).orElseThrow();
+        m.addAttribute("profileusername", profileUser.getUsername());
+        m.addAttribute("profileId", profileUser.getId());
+        m.addAttribute("profileDateCreated", profileUser.getDateCreated());
+        return "user-info.html";
+
+    }
+
+
+    @PutMapping("/user/{id}")
+    public RedirectView updateUserInfo(Model m, Principal p, @PathVariable Long id, String profileUsername,
+                                       RedirectAttributes redir) {
+        SiteUser userToBeEdited = siteUserRepository.findById(id).orElseThrow();
```

Comment:
These should all be named to match your own variables - `SiteUser` should be `AppUser`, `siteUserRepository` should be `appUserRepository`, etc.